docs/prover/checking/sanity.md

```diff
@@ -2,9 +2,241 @@ Rule Sanity Checks
 ==================
 
 The {ref}`--rule_sanity` option enables some automatic checks that can warn you
-about certain classes of mistakes in specifications.
+about certain classes of mistakes in specifications. The `—rule_sanity` options 
+may be followed by one of `none`, `basic`, or `advanced` options.
+
+There are 3 kinds of sanity checks:
+
+1. **Reachability** checks that even when ignoring all the user-provided
+   assertions, the end of the rule is reachable. This check ensures that that
+   the combination of `require` statements does not rule out all possible
+   counterexamples.
+
+   For example, the following rule would be flagged by the reachability check:
+   ```cvl
+   rule vacuous {
+     uint x;
+     require x > 2;
+     require x < 1;
+     assert f(x) == 2, "f must return 2";
+   }
+   ```
+   Since there are no models satisfying both `x > 2` and `x < 1`, this rule
+   will always pass, regardless of the behavior of the contract.  This is an
+   example of a *vacuous* rule - one that passes only because the preconditions
+   are contradictory.
+
+   ```{caution}
+   The reachability check will *pass* on vacuous rules and *fail* on correct
+   rules.  A passing reachability check indicates a potential error in the rule.
+   
+   The exception is when a {term}`parametric rule` is checked on the default
+   fallback function: The default fallback function should always revert, so
+   there are no examples that can reach the end of the rule.
+   ```
+   
+2. **Assert-Vacuity** checks that individual `assert` statements are not
+   tautologies.  A tautology is a statement that is true on all examples, even
```

Comment:
A tautology is defined as a statement that is true on all examples, even
   if all the `require` and `if` conditions are removed.